core/src/test/java/org/apache/iceberg/TestPartitioning.java

```diff
@@ -140,9 +140,9 @@ public void testPartitionTypeWithAddingBackSamePartitionFieldInV1Table() {
         .addField("data")
         .commit();
 
-    // in v1, we use void transforms instead of dropping partition fields
+    // in v1, we use void transforms instead of dropping partition fields.
     StructType expectedType = StructType.of(
-        NestedField.optional(1000, "data_1000", Types.StringType.get()),
+        NestedField.optional(1000, "data", Types.StringType.get()),
```

Comment:
This change is wrong. The previous expected value was correct as we need to inherit the newest column names.

Reply:
reverted